### PR TITLE
Create pty host starter interface and move pty host to main process

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #175335

### CODE REVIEW

**Comment by Tyriar** [comment]
Where I ended up getting on this:

- Shuffled some things around to support `IPtyHostStarter`
- On desktop, moved the pty host to under the main process, this ended up causing this issue with conpty which I got stuck on https://github.com/microsoft/vscode/issues/175335#issuecomment-1491990166

**Comment by Tyriar** [comment]
The conpty problem doesn't seem to be happening anymore 🤷

File: src/vs/platform/terminal/node/ptyHostService.ts [hunk 1]
```diff
@@ -191,7 +163,13 @@ export class PtyHostService extends Disposable implements IPtyService {
 		this._register(proxy.onProcessOrphanQuestion(e => this._onProcessOrphanQuestion.fire(e)));
 		this._register(proxy.onDidRequestDetach(e => this._onDidRequestDetach.fire(e)));
 
-		return [client, proxy];
+		// HACK: When RemoteLoggerChannelClient is not delayed, the Pty Host log file won't show up
+		// in the Output view of the first window?
+		Event.once(Event.any(proxy.onProcessReady, proxy.onProcessReplay))(() => {
+			this._register(new RemoteLoggerChannelClient(this._loggerService, client.getChannel(TerminalIpcChannels.Logger)));
+		});
```
Comment (Tyriar) [inline src/vs/platform/terminal/node/ptyHostService.ts:170]: Not sure why but logging will not work without delaying it like this, it seems to register the logger when called regardless, it just doesn't show up in the first window's output view 😕

**Comment by Tyriar** [comment]
Finally past logging, looking into why remote terminals aren't working now

**Comment by Tyriar** [comment]
Web remote terminals work, but the desktop pty host process exits almost immediately

**Comment by Tyriar** [comment]
It's conpty again, terminals work everywhere when it's disabled, but it somehow takes down the remote pty host in the desktop case only now (not web/server)

**Comment by Tyriar** [comment]
The test resolver (doesn't work) and the web target (does work), both seemt o use the same version of node (`/.build/node/...` via code-server.bat).

**Comment by Tyriar** [comment]
Still stuck on this conpty thing, I'm doing a product build to see if the issue is there too, if not I'm going to move ahead with this.

**Comment by Tyriar** [comment]
It seems to work, if there is an issue it would be restricted to remote Windows which I'm not sure how to set up locally.

File: src/vs/platform/terminal/electron-main/electronPtyHostStarter.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		// 	}
+		// };
+
+		// const ptyHostDebug = parsePtyHostDebugPort(this._environmentService.args, this._environmentService.isBuilt);
+		// if (ptyHostDebug) {
+		// 	if (ptyHostDebug.break && ptyHostDebug.port) {
+		// 		opts.debugBrk = ptyHostDebug.port;
+		// 	} else if (!ptyHostDebug.break && ptyHostDebug.port) {
+		// 		opts.debug = ptyHostDebug.port;
+		// 	}
```
Comment (TylerLeonhardt) [inline src/vs/platform/terminal/electron-main/electronPtyHostStarter.ts:52]: commented out code

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**